src/logcollector/config.c

```diff
@@ -101,162 +101,199 @@ int LogCollectorConfig(const char *cfgfile)
     return (1);
 }
 
-
-void _getLocalfilesListJSON(logreader *list, cJSON *array, int gl) {
-
-    unsigned int i = 0;
-    unsigned int j;
-
-    while ((!gl && list[i].target) || (gl && list[i].file)) {
-        cJSON *file = cJSON_CreateObject();
-
-        if (list[i].file) cJSON_AddStringToObject(file,"file",list[i].file);
-        if (list[i].channel_str != NULL) cJSON_AddStringToObject(file, "channel", list[i].channel_str);
-        if (list[i].logformat) cJSON_AddStringToObject(file,"logformat",list[i].logformat);
-        if (list[i].command) cJSON_AddStringToObject(file,"command",list[i].command);
-        if (list[i].djb_program_name) cJSON_AddStringToObject(file,"djb_program_name",list[i].djb_program_name);
-        if (list[i].alias) cJSON_AddStringToObject(file,"alias",list[i].alias);
-        if (list[i].query != NULL) {
-            cJSON * query = cJSON_CreateObject();
-            if (*list[i].query != '\0') {
-                cJSON_AddStringToObject(query, "value", list[i].query);
+void _getLocalfilesListJSON(logreader* reader, cJSON* array, const char *gpath)
+{
+    cJSON* file = cJSON_CreateObject();
+
+    if (gpath)
+        cJSON_AddStringToObject(file, "file", gpath);
```

Comment:
the "file" field should not be named "location"? to maintain consistency with the configuration.

Reply:
@cborla If we were designing this from scratch, I'd say yes. But since this is a port, we prefer to avoid making changes that could be disruptive, whether in the API, Dashboard, or third-party applications.